internal/completion/complete.go

```diff
@@ -0,0 +1,316 @@
+
+			var key interface{}
+			var keyStr string
+			if flag != nil {
+				key = flag
+				keyStr = flag.Name
+			} else {
+				key = finalCmd
+				keyStr = finalCmd.Name()
+			}
+
+			// Find completion function for the flag or command
+			completionFn, ok := validArgsFunctions[key]
+			if !ok {
+				CompErrorln(fmt.Sprintf("Dynamic completion not supported/needed for flag or command: %s", keyStr))
+				return
+			}
+
+			CompDebugln(fmt.Sprintf("Calling completion method for subcommand '%s' with args '%v' and toComplete '%s'", finalCmd.Name(), argsWoFlags, toComplete))
+			completions, directive := completionFn(finalCmd, argsWoFlags, toComplete)
+			for _, comp := range completions {
+				// Print each possible completion to stdout for the completion script to consume.
+				fmt.Println(comp)
+			}
+
+			// Print some helpful info to stderr for the user to see.
+			// Output from stderr should be ignored from the completion script.
+			fmt.Fprintf(os.Stderr, "Completion ended with directive: %s\n", directive.string())
+			os.Exit(int(directive))
+		},
+	}
+}
+
+func isFlag(arg string) bool {
+	return len(arg) > 0 && arg[0] == '-'
+}
+
+func checkIfFlagCompletion(rootCmd *cobra.Command, args []string, lastArg string) (*pflag.Flag, []string, string) {
+	var flagName string
+	trimmedArgs := args
+	flagWithEqual := false
+	if isFlag(lastArg) {
+		if index := strings.Index(lastArg, "="); index >= 0 {
+			flagName = strings.TrimLeft(lastArg[:index], "-")
+			lastArg = lastArg[index+1:]
+			flagWithEqual = true
+		} else {
+			CompErrorln("Unexpected completion request for flag")
+			os.Exit(int(BashCompDirectiveError))
+		}
+	}
+
+	if len(flagName) == 0 {
+		if len(args) > 0 {
+			prevArg := args[len(args)-1]
+			if isFlag(prevArg) {
+				// If the flag contains an = it means it has already been fully processed
+				if index := strings.Index(prevArg, "="); index < 0 {
+					flagName = strings.TrimLeft(prevArg, "-")
+
+					// Remove the uncompleted flag or else Cobra could complain about
+					// an invalid value for that flag e.g., helm status --output j<TAB>
+					trimmedArgs = args[:len(args)-1]
+				}
+			}
+		}
+	}
+
+	if len(flagName) == 0 {
+		// Not doing flag completion
+		return nil, trimmedArgs, lastArg
+	}
+
+	// Find the real command for which completion must be performed
+	finalCmd, _, err := rootCmd.Find(trimmedArgs)
+	if err != nil {
+		// Unable to find the real command. E.g., helm invalidCmd <TAB>
+		os.Exit(int(BashCompDirectiveError))
+	}
+
+	CompDebugln(fmt.Sprintf("checkIfFlagCompletion: found final command '%s'", finalCmd.Name()))
+
+	flag := findFlag(finalCmd, flagName)
+	if flag == nil {
+		// Flag not supported by this command, nothing to complete
+		CompDebugln(fmt.Sprintf("Subcommand '%s' does not support flag '%s'", finalCmd.Name(), flagName))
+		os.Exit(int(BashCompDirectiveNoFileComp))
+	}
+
+	if !flagWithEqual {
+		if _, err = finalCmd.Flags().GetBool(flag.Name); err == nil {
```

Comment:
This code is wrong.  I won't work for inherited flags.  So the following completion will not trigger the dynamic completion:
helm __complete status --debug ""

Reply:
I have forced pushed a fix for this.